### ISSUE TITLE
Screen readers do not announce diffs

### ISSUE THREAD

**Issue by jayspadie** [Microsoft/monaco-editor#445]
monaco-editor version: 0.8.3
Browser: Any
OS: Windows 10

During accessibility testing, it was found that screen readers do not announce diff information.  This is a pretty broad issue, but the ideal solution might include such things as:
* Upon entering the diff editor, reporting the total number of lines added, removed, and changed
* Native/builtin keyboard navigation to prev/next diff
* Line numbers can be heard
* While arrowing down lines, reporting if it was added, removed, or edited (including the line content).

Shown below is what NVDA announces with Firefox while in the modified pane.  It announces the line content while arrowing down, but there is no indication of added, removed, or edited line information.

![image](https://cloud.githubusercontent.com/assets/5246869/26265776/736d0134-3cb0-11e7-9fb9-041d94d413c9.png)

**Comment by alexandrudima** [comment]
I think in this case we need to build a custom experience, similar to F8 which loops through errors/warnings. The diff editor, like a regular editor, has the role "textbox" and it can be editable or not. We cannot change what screen readers announce when the cursor changes to a different line, since we cannot annotate in any way a "textbox"'s lines with additional properties, since our "textbox" <-> textarea.

To investigate:
* implementing a custom experience that loops through diffs and can announce both old lines and new lines
* moving away from a textarea to using an iframe with content editable for I/O.

**Comment by alexandrudima** [comment]
Let' continue tracking in Microsoft/vscode#17263
